Client/Frontend/Browser/BraveSearchHelper.swift

```diff
@@ -0,0 +1,116 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import WebKit
+import Shared
+import BraveShared
+
+private let log = Logger.browserLogger
+
+class BraveSearchHelper: TabContentScript {
+    private weak var tab: Tab?
+    private let profile: Profile
+    
+    /// Tracks how many in current browsing session the user has been prompted to set Brave Search as a default
+    /// while on one of Brave Search websites.
+    private static var canSetAsDefaultCounter = 0
```

Comment:
Why is this value static ? Instead I believe either we should store it as a preference or have this value in bvc.

Reply:
This stores per-session value, which does not need to be persisted.
I like having it here because it is used in this file only and we can keep it `private`